From e60cecc20519c6d5f1547ea2a5e3ee3ae1fd3966 Mon Sep 17 00:00:00 2001
From: Ethan Chen <intervigil@gmail.com>
Date: Sat, 22 Jun 2019 14:12:40 -0700
Subject: [PATCH 32/36] build: Separate platform macros from QCOM platform
 definition

* The macros used in the file formerly called "qcom_utils.mk" are
  actually generic and not QCOM-specific. Move the QCOM_BOARD definitions
  to a QCOM-specific location, and name the remaning macros something more
  generic.

Change-Id: I6747bc9201e56b456caf251122057d7dbc233f1c
---
 build/core/config.mk                          |  4 +-
 build/core/definitions.mk                     |  7 ++-
 build/core/mtk_target.mk                      |  2 -
 build/core/{qcom_utils.mk => utils.mk}        | 49 -------------------
 .../{mtk_utils.mk => vendor/mtk_boards.mk}    |  0
 build/core/vendor/qcom_boards.mk              | 43 ++++++++++++++++
 6 files changed, 50 insertions(+), 55 deletions(-)
 mode change 100644 => 100755 build/core/config.mk
 delete mode 100644 build/core/mtk_target.mk
 rename build/core/{qcom_utils.mk => utils.mk} (85%)
 mode change 100644 => 100755
 rename build/core/{mtk_utils.mk => vendor/mtk_boards.mk} (100%)
 create mode 100644 build/core/vendor/qcom_boards.mk

diff --git a/build/core/config.mk b/build/core/config.mk
old mode 100644
new mode 100755
index 2f758c97..3bf8e2c0
--- a/build/core/config.mk
+++ b/build/core/config.mk
@@ -22,5 +22,5 @@ FRAMEWORK_LINEAGE_API_NEEDS_UPDATE_TEXT := $(TOPDIR)vendor/lineage/build/core/ap
 
 BUILD_RRO_SYSTEM_PACKAGE := $(TOPDIR)vendor/lineage/build/core/system_rro.mk
 
-# Rules for MTK targets
-include $(TOPDIR)vendor/lineage/build/core/mtk_target.mk
+# Rules for QCOM targets
+include $(TOPDIR)vendor/lineage/build/core/qcom_target.mk
diff --git a/build/core/definitions.mk b/build/core/definitions.mk
index 1cd5a92f..82ed55d4 100644
--- a/build/core/definitions.mk
+++ b/build/core/definitions.mk
@@ -5,5 +5,8 @@ define uniq
 $(if $1,$(firstword $1) $(call uniq,$(filter-out $(firstword $1),$1)))
 endef
 
-# Include QCOM macros
-include vendor/lineage/build/core/qcom_utils.mk
+# Include board/platform macros
+include vendor/lineage/build/core/utils.mk
+
+# Include vendor platform definitions
+include vendor/lineage/build/core/vendor/*.mk
diff --git a/build/core/mtk_target.mk b/build/core/mtk_target.mk
deleted file mode 100644
index 576e3833..00000000
--- a/build/core/mtk_target.mk
+++ /dev/null
@@ -1,2 +0,0 @@
-# Bring in Mediatek helper macros
-include vendor/lineage/build/core/mtk_utils.mk
diff --git a/build/core/qcom_utils.mk b/build/core/utils.mk
old mode 100644
new mode 100755
similarity index 85%
rename from build/core/qcom_utils.mk
rename to build/core/utils.mk
index 7811f88e..ae1db16c
--- a/build/core/qcom_utils.mk
+++ b/build/core/utils.mk
@@ -1,52 +1,3 @@
-# Board platforms lists to be used for
-# TARGET_BOARD_PLATFORM specific featurization
-
-# A Family
-QCOM_BOARD_PLATFORMS += msm7x27a
-QCOM_BOARD_PLATFORMS += msm7x30
-QCOM_BOARD_PLATFORMS += msm8660
-
-QCOM_BOARD_PLATFORMS += msm8960
-
-# B Family
-QCOM_BOARD_PLATFORMS += msm8226
-QCOM_BOARD_PLATFORMS += msm8610
-QCOM_BOARD_PLATFORMS += msm8974
-
-QCOM_BOARD_PLATFORMS += apq8084
-
-# B64 Family
-QCOM_BOARD_PLATFORMS += msm8992
-QCOM_BOARD_PLATFORMS += msm8994
-
-# BR Family
-QCOM_BOARD_PLATFORMS += msm8909
-QCOM_BOARD_PLATFORMS += msm8916
-
-QCOM_BOARD_PLATFORMS += msm8952
-
-# UM Family
-QCOM_BOARD_PLATFORMS += msm8937
-QCOM_BOARD_PLATFORMS += msm8953
-QCOM_BOARD_PLATFORMS += msm8996
-
-QCOM_BOARD_PLATFORMS += msm8998
-QCOM_BOARD_PLATFORMS += sdm660
-
-QCOM_BOARD_PLATFORMS += sdm845
-
-QCOM_BOARD_PLATFORMS += sm6150
-QCOM_BOARD_PLATFORMS += sm8150
-
-# MSM7000 Family
-MSM7K_BOARD_PLATFORMS := msm7x30
-MSM7K_BOARD_PLATFORMS += msm7x27
-MSM7K_BOARD_PLATFORMS += msm7x27a
-MSM7K_BOARD_PLATFORMS += msm7k
-
-QSD8K_BOARD_PLATFORMS := qsd8k
-
-
 # vars for use by utils
 empty :=
 space := $(empty) $(empty)
diff --git a/build/core/mtk_utils.mk b/build/core/vendor/mtk_boards.mk
similarity index 100%
rename from build/core/mtk_utils.mk
rename to build/core/vendor/mtk_boards.mk
diff --git a/build/core/vendor/qcom_boards.mk b/build/core/vendor/qcom_boards.mk
new file mode 100644
index 00000000..4fcb0db7
--- /dev/null
+++ b/build/core/vendor/qcom_boards.mk
@@ -0,0 +1,43 @@
+# Board platforms lists to be used for
+# TARGET_BOARD_PLATFORM specific featurization
+
+# A Family
+QCOM_BOARD_PLATFORMS += msm7x27a
+QCOM_BOARD_PLATFORMS += msm7x30
+QCOM_BOARD_PLATFORMS += msm8660
+
+QCOM_BOARD_PLATFORMS += msm8960
+
+# B Family
+QCOM_BOARD_PLATFORMS += msm8226
+QCOM_BOARD_PLATFORMS += msm8610
+QCOM_BOARD_PLATFORMS += msm8974
+
+QCOM_BOARD_PLATFORMS += apq8084
+
+# B64 Family
+QCOM_BOARD_PLATFORMS += msm8992
+QCOM_BOARD_PLATFORMS += msm8994
+
+# BR Family
+QCOM_BOARD_PLATFORMS += msm8909
+QCOM_BOARD_PLATFORMS += msm8916
+
+QCOM_BOARD_PLATFORMS += msm8952
+
+# UM Family
+QCOM_BOARD_PLATFORMS += msm8937
+QCOM_BOARD_PLATFORMS += msm8953
+QCOM_BOARD_PLATFORMS += msm8996
+QCOM_BOARD_PLATFORMS += msm8998
+
+QCOM_BOARD_PLATFORMS += sdm660
+QCOM_BOARD_PLATFORMS += sdm845
+
+# MSM7000 Family
+MSM7K_BOARD_PLATFORMS := msm7x30
+MSM7K_BOARD_PLATFORMS += msm7x27
+MSM7K_BOARD_PLATFORMS += msm7x27a
+MSM7K_BOARD_PLATFORMS += msm7k
+
+QSD8K_BOARD_PLATFORMS := qsd8k
-- 
2.17.1

